src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,67 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { useEffect, useState } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  onFilterSelected: (nameFilter: string) => void;
+  textInput: (text: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  onFilterSelected, textInput,
+}) => {
+  const [statusSelect, setStatusSelect] = useState('');
+  const [input, setInput] = useState('');
+
+  useEffect(() => {
+    onFilterSelected(statusSelect);
+  }, [statusSelect]);
+
+  useEffect(() => {
+    textInput(input.toLowerCase());
+  }, [input]);
+
```

Comment:
I don't understand why this is needed? When the state changes, the re-render will happen anyway.